crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -618,6 +604,94 @@ impl Debug for UnsafeWorldCell<'_> {
     }
 }
 
+/// Gives read-only access to a [`World`]'s metadata.
+#[derive(Clone, Copy)]
```

Comment:
Follow-up: can we make this a `SystemParam` and `ExclusiveSystemParam` too?

Reply:
I don't think so, since an exclusive system takes a `&mut World` and that has mutable access to the metadata. For this same reason `&Archetypes`, `&Entities` etc don't implement `ExclusiveSystemParam` but only `SystemParam`.